runelite-api/src/main/java/net/runelite/api/VarPlayer.java

```diff
@@ -231,6 +231,36 @@
 	 */
 	LAST_MINIGAME_TELEPORT(888),
 
+	/**
+	 * The location of the player's dwarf cannon
+	 * <p>
+	 * X-Coordinate = (Value % 128^4) / 128^2<br>
+	 * Y-Coordinate = Value % 128^2<br>
+	 * Z-Coordinate = Value / 128^4
```

Comment:
While this is an accurate way to describe it, it's not really the best way to think about these values; the values transmitted are bit-packed so that all three coordinate values are stored in a single variable. `128 ^ 4` (which is `268_435_456`) is better represented via bitwise operations as `1 << 28` and `128 ^ 2` as `1 << 14`. All these divisions and modulo operations are similarly able to be represented as bitwise operations:

* x: `(val & ((1 << 28) - 1)) >> 14`
* y: `val & ((1 << 14) -1)`
* z: `val >> 28`

As such, this javadoc (and the code interfacing it) would make much more sense if it were described with that knowledge in mind.

In this case, you could describe `DWARF_CANNON_LOCATION` as a bit-packed int of the following format:

```java
/**
  * The location of the player's dwarf cannon, represented by a bit-packed int of the following format:
  * <p>
  * 0zzzxxxxxxxxxxxxxxyyyyyyyyyyyyyy<br>
  * (that is, z represented by the first 3 positive bits followed by x and y each represented by 14 bits)
  * <p>
  * Getting the value of each of these coordinates can be accomplished [...]
```